pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -196,22 +198,26 @@ def _on_evaluation_model_train(self) -> None:
     def _on_evaluation_end(self, *args: Any, **kwargs: Any) -> None:
         """Runs ``on_{validation/test}_end`` hook."""
         if self.trainer.testing:
-            self.trainer.call_hook("on_test_end", *args, **kwargs)
+            self.trainer._call_hook(self.trainer, "on_test_end", *args, **kwargs)
+            self.trainer._call_hook(self.trainer.lightning_module, "on_test_end", *args, **kwargs)
         else:
-            self.trainer.call_hook("on_validation_end", *args, **kwargs)
+            self.trainer._call_hook(self.trainer.lightning_module, "on_validation_end", *args, **kwargs)
 
         # reset the logger connector state
         self.trainer.logger_connector.reset_results()
 
     def _on_evaluation_epoch_start(self, *args: Any, **kwargs: Any) -> None:
         """Runs ``on_epoch_start`` and ``on_{validation/test}_epoch_start`` hooks."""
         self.trainer.logger_connector.on_epoch_start()
-        self.trainer.call_hook("on_epoch_start", *args, **kwargs)
+        self.trainer._call_hook(self.trainer, "on_epoch_start", *args, **kwargs)
+        self.trainer._call_hook(self.trainer.lightning_module, "on_epoch_start", *args, **kwargs)
 
         if self.trainer.testing:
-            self.trainer.call_hook("on_test_epoch_start", *args, **kwargs)
+            self.trainer._call_hook(self.trainer, "on_test_epoch_start", *args, **kwargs)
+            self.trainer._call_hook(self.trainer.lightning_module, "on_test_epoch_start", *args, **kwargs)
         else:
-            self.trainer.call_hook("on_validation_epoch_start", *args, **kwargs)
+            self.trainer._call_hook(self.trainer, "on_validation_epoch_start", *args, **kwargs)
```

Comment:
what should I do in the case of hooks like `on_validation_epoch_start` where the hook in callbacks takes trainer + LM, but the hook on LM takes no input args?

Reply:
That's managed already by the `TrainerCallbackHookMixin`